File: 6___Data_And_Matplotlib/SpeedTestPythonVsC/speedTest.cc
#include <iostream>
#include <chrono>

int main() {
    const long iterations = 1000000000; // 1 milliard og 100 mio iterationer
    long sum = 0;

    auto start = std::chrono::high_resolution_clock::now();

    for (long i = 0; i < iterations; ++i) {
        sum += i;
    }

    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> elapsed = end - start;

    std::cout << "Elapsed time: " << elapsed.count() << " seconds\n";
    std::cout << "Sum: " << sum << std::endl;

    return 0;
}

File: 6___Data_And_Matplotlib/SpeedTestPythonVsC/speedTest.py
import time

iterations = 1000000000  # 1 milliard, 100 mio, 10 mio iterationer
sum = 0

start = time.perf_counter()  # starttid i sekunder med høj opløsning

for i in range(iterations):
    sum += i

end = time.perf_counter()  # sluttid
elapsed = end - start

print("Elapsed time:", elapsed, "seconds")
print("Sum:", sum)